tests/Microsoft.Identity.Test.Unit.net45/Resources/ExpectedTokenCache.json

```diff
@@ -0,0 +1,201 @@
+      "client_info": null,
+      "client_id": "d3adb33f-c0de-ed0c-c0de-deadb33fc0d3",
+      "secret": "access_token_secret",
+      "credential_type": "RefreshToken",
+      "family_id": "1"
+    }
+  },
+  "IdToken": {
+    "my-uid.my-utid-env_1-idtoken-d3adb33f-c0de-ed0c-c0de-deadb33fc0d3-the_tenant_id-": {
+      "home_account_id": "my-uid.my-utid",
+      "environment": "env_1",
+      "client_info": null,
+      "client_id": "d3adb33f-c0de-ed0c-c0de-deadb33fc0d3",
+      "secret": "access_token_secret",
+      "credential_type": "IdToken",
+      "realm": "the_tenant_id"
+    },
+    "my-uid.my-utid-env_2-idtoken-d3adb33f-c0de-ed0c-c0de-deadb33fc0d3-the_tenant_id-": {
+      "home_account_id": "my-uid.my-utid",
+      "environment": "env_2",
+      "client_info": null,
+      "client_id": "d3adb33f-c0de-ed0c-c0de-deadb33fc0d3",
+      "secret": "access_token_secret",
+      "credential_type": "IdToken",
+      "realm": "the_tenant_id"
+    },
+    "my-uid.my-utid-env_3-idtoken-d3adb33f-c0de-ed0c-c0de-deadb33fc0d3-the_tenant_id-": {
+      "home_account_id": "my-uid.my-utid",
+      "environment": "env_3",
+      "client_info": null,
+      "client_id": "d3adb33f-c0de-ed0c-c0de-deadb33fc0d3",
+      "secret": "access_token_secret",
+      "credential_type": "IdToken",
+      "realm": "the_tenant_id"
+    }
+  },
+  "Account": {
+    "my-uid.my-utid-env_1-the_tenant_id": {
+      "home_account_id": "my-uid.my-utid",
+      "environment": "env_1",
+      "client_info": null,
+      "username": "joe@localhost.com",
+      "name": "First Last",
+      "given_name": "Joe",
+      "family_name": "Doe",
+      "local_account_id": "test_local_account_id",
+      "authority_type": "authority type",
+      "realm": "the_tenant_id"
+    },
+    "my-uid.my-utid-env_2-the_tenant_id": {
+      "home_account_id": "my-uid.my-utid",
+      "environment": "env_2",
+      "client_info": null,
+      "username": "joe@localhost.com",
+      "name": "First Last",
+      "given_name": "Joe",
+      "family_name": "Doe",
+      "local_account_id": "test_local_account_id",
+      "authority_type": "authority type",
+      "realm": "the_tenant_id"
+    },
+    "my-uid.my-utid-env_3-the_tenant_id": {
+      "home_account_id": "my-uid.my-utid",
+      "environment": "env_3",
+      "client_info": null,
+      "username": "joe@localhost.com",
+      "name": "First Last",
+      "given_name": "Joe",
+      "family_name": "Doe",
+      "local_account_id": "test_local_account_id",
+      "authority_type": "authority type",
+      "realm": "the_tenant_id"
+    }
+  },
+  "AppMetadata": {
+    "appmetadata-env_1-d3adb33f-c0de-ed0c-c0de-deadb33fc0d3": {
+      "environment": "env_1",
+      "client_id": "d3adb33f-c0de-ed0c-c0de-deadb33fc0d3",
+      "family_id": "1"
+    },
+    "appmetadata-env_2-d3adb33f-c0de-ed0c-c0de-deadb33fc0d3": {
+      "environment": "env_2",
+      "client_id": "d3adb33f-c0de-ed0c-c0de-deadb33fc0d3",
+      "family_id": ""
```

Comment:
The "orphan" RT's family_id value is `null` above, but its value is empty string `""` here. Is that intentional?
(It is not necessarily an issue, considering that the deserializing part is supposed to handle both. Just curious to ask. :-) )

Reply:
It is intentional because I created this via a test :) so I was creating familyId as null or empty 

In production I expect only empty strings, but as you say, gotta be prepared for both null and "".